app/clients/ethereum/ethereum.go

```diff
@@ -131,3 +135,31 @@ func (ec *Client) waitForTransactionReceipt(hash common.Hash) (txReceipt *types.
 
 	return ec.Client.TransactionReceipt(context.Background(), hash)
 }
+
+func (ec *Client) WaitForConfirmations(raw types.Log) error {
+	target := raw.BlockNumber + ec.config.BlockConfirmations
+	for {
+		currentBlockNumber, err := ec.BlockNumber(context.Background())
+		if err != nil {
+			ec.logger.Errorf("[%s] Failed retrieving block number.", raw.TxHash.String())
+			return err
+		}
+
+		if target <= currentBlockNumber {
+			receipt, err := ec.TransactionReceipt(context.Background(), raw.TxHash)
+			if err != nil {
```

Comment:
Lets parse the 2 types of errors -> if not found or any other.
If not found -> `[%s] Ethereum TX went into an uncle block` with `debug` level of the log
if any other -> `[%s] Failed to get Transaction receipt` and return `err`